### ISSUE TITLE
`text-1.2.0.3` incorrectly claims to support `base < 4.5` && `bytestring >= 0.10.4`

### ISSUE THREAD

**Issue by hvr** [bos/text#108]
The error that occurs then:

```
Data/Text/Lazy/Encoding.hs:65:29:
     Module `Data.Monoid' does not export `(<>)'
```

The problem is that `(<>)` exists only since `base-4.5`

The easiest way to fix this is to add

```
  if !impl(ghc >= 7.4)
    build-depends: bytestring < 0.10.4
```

to the `text.cabal` file to keep the cabal solver from running into that combination.

**Comment by bos** [comment]
Can you give me some more details? I build the text package with every release of GHC from 7.0 on, and I haven't seen this occur.

**Comment by hvr** [comment]
Another example is a mere `cabal install scientific` which is prevented from selecting a working install-plan:

```
$ cabal install -w /opt/ghc/7.2.2/bin/ghc scientific -j
Resolving dependencies...
Notice: installing into a sandbox located at /tmp/sb1/.cabal-sandbox
Configuring deepseq-1.3.0.2...
Building deepseq-1.3.0.2...
Installed deepseq-1.3.0.2
Configuring bytestring-0.10.4.1...
Building bytestring-0.10.4.1...
Installed bytestring-0.10.4.1
Configuring text-1.2.0.3...
Building text-1.2.0.3...
Failed to install text-1.2.0.3
Build log ( /tmp/sb1/.cabal-sandbox/logs/text-1.2.0.3.log ):
Configuring text-1.2.0.3...
Building text-1.2.0.3...
Preprocessing library text-1.2.0.3...
[ 1 of 43] Compiling Data.Text.Internal.Read ( Data/Text/Internal/Read.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Read.o )
[ 2 of 43] Compiling Data.Text.Internal.Encoding.Utf32 ( Data/Text/Internal/Encoding/Utf32.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Encoding/Utf32.o )
[ 3 of 43] Compiling Data.Text.Internal.Builder.RealFloat.Functions ( Data/Text/Internal/Builder/RealFloat/Functions.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Builder/RealFloat/Functions.o )
[ 4 of 43] Compiling Data.Text.Internal.Builder.Int.Digits ( Data/Text/Internal/Builder/Int/Digits.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Builder/Int/Digits.o )
[ 5 of 43] Compiling Data.Text.Encoding.Error ( Data/Text/Encoding/Error.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Encoding/Error.o )
[ 6 of 43] Compiling Data.Text.Internal.Unsafe.Shift ( Data/Text/Internal/Unsafe/Shift.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Unsafe/Shift.o )
[ 7 of 43] Compiling Data.Text.Internal.Encoding.Utf16 ( Data/Text/Internal/Encoding/Utf16.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Encoding/Utf16.o )
[ 8 of 43] Compiling Data.Text.Internal.Functions ( Data/Text/Internal/Functions.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Functions.o )
[ 9 of 43] Compiling Data.Text.Internal.Unsafe ( Data/Text/Internal/Unsafe.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Unsafe.o )
[10 of 43] Compiling Data.Text.Array  ( Data/Text/Array.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Array.o )
[11 of 43] Compiling Data.Text.Internal.Unsafe.Char ( Data/Text/Internal/Unsafe/Char.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Unsafe/Char.o )
[12 of 43] Compiling Data.Text.Internal ( Data/Text/Internal.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal.o )
[13 of 43] Compiling Data.Text.Internal.Fusion.Size ( Data/Text/Internal/Fusion/Size.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Fusion/Size.o )
[14 of 43] Compiling Data.Text.Internal.Fusion.Types ( Data/Text/Internal/Fusion/Types.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Fusion/Types.o )
[15 of 43] Compiling Data.Text.Internal.Fusion.CaseMapping ( Data/Text/Internal/Fusion/CaseMapping.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Fusion/CaseMapping.o )
[16 of 43] Compiling Data.Text.Internal.Fusion.Common ( Data/Text/Internal/Fusion/Common.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Fusion/Common.o )
[17 of 43] Compiling Data.Text.Unsafe ( Data/Text/Unsafe.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Unsafe.o )
[18 of 43] Compiling Data.Text.Internal.Private ( Data/Text/Internal/Private.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Private.o )
[19 of 43] Compiling Data.Text.Internal.Fusion ( Data/Text/Internal/Fusion.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Fusion.o )
[20 of 43] Compiling Data.Text.Internal.Encoding.Fusion.Common ( Data/Text/Internal/Encoding/Fusion/Common.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Encoding/Fusion/Common.o )
[21 of 43] Compiling Data.Text.Internal.Encoding.Utf8 ( Data/Text/Internal/Encoding/Utf8.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Encoding/Utf8.o )
[22 of 43] Compiling Data.Text.Internal.Encoding.Fusion ( Data/Text/Internal/Encoding/Fusion.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Encoding/Fusion.o )
[23 of 43] Compiling Data.Text.Internal.Lazy.Encoding.Fusion ( Data/Text/Internal/Lazy/Encoding/Fusion.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Lazy/Encoding/Fusion.o )
[24 of 43] Compiling Data.Text.Internal.Search ( Data/Text/Internal/Search.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Search.o )
[25 of 43] Compiling Data.Text        ( Data/Text.hs, dist/dist-sandbox-4ac2267c/build/Data/Text.o )
[26 of 43] Compiling Data.Text.Encoding ( Data/Text/Encoding.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Encoding.o )

Data/Text/Encoding.hs:95:1:
    Warning: The import item `CSize(..)' suggests that
             `CSize' has (in-scope) constructors or class methods,
             but it has none
[27 of 43] Compiling Data.Text.Foreign ( Data/Text/Foreign.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Foreign.o )
[28 of 43] Compiling Data.Text.Internal.IO ( Data/Text/Internal/IO.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/IO.o )
[29 of 43] Compiling Data.Text.IO     ( Data/Text/IO.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/IO.o )
[30 of 43] Compiling Data.Text.Internal.Lazy ( Data/Text/Internal/Lazy.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Lazy.o )
[31 of 43] Compiling Data.Text.Internal.Lazy.Fusion ( Data/Text/Internal/Lazy/Fusion.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Lazy/Fusion.o )
[32 of 43] Compiling Data.Text.Internal.Lazy.Search ( Data/Text/Internal/Lazy/Search.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Lazy/Search.o )
[33 of 43] Compiling Data.Text.Lazy.Internal ( Data/Text/Lazy/Internal.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Lazy/Internal.o )
[34 of 43] Compiling Data.Text.Lazy   ( Data/Text/Lazy.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Lazy.o )
[35 of 43] Compiling Data.Text.Internal.Builder ( Data/Text/Internal/Builder.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Builder.o )
[36 of 43] Compiling Data.Text.Lazy.Builder ( Data/Text/Lazy/Builder.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Lazy/Builder.o )
[37 of 43] Compiling Data.Text.Internal.Builder.Functions ( Data/Text/Internal/Builder/Functions.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Internal/Builder/Functions.o )
[38 of 43] Compiling Data.Text.Lazy.Builder.Int ( Data/Text/Lazy/Builder/Int.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Lazy/Builder/Int.o )
[39 of 43] Compiling Data.Text.Lazy.IO ( Data/Text/Lazy/IO.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Lazy/IO.o )
[40 of 43] Compiling Data.Text.Lazy.Read ( Data/Text/Lazy/Read.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Lazy/Read.o )
[41 of 43] Compiling Data.Text.Lazy.Builder.RealFloat ( Data/Text/Lazy/Builder/RealFloat.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Lazy/Builder/RealFloat.o )
[42 of 43] Compiling Data.Text.Lazy.Encoding ( Data/Text/Lazy/Encoding.hs, dist/dist-sandbox-4ac2267c/build/Data/Text/Lazy/Encoding.o )

Data/Text/Lazy/Encoding.hs:65:29:
    Module `Data.Monoid' does not export `(<>)'
cabal: Error: some packages failed to install:
hashable-1.2.3.1 depends on text-1.2.0.3 which failed to install.
scientific-0.3.3.5 depends on text-1.2.0.3 which failed to install.
text-1.2.0.3 failed during the building phase. The exception was:
ExitFailure 1
```

**Comment by bos** [comment]
Released as 1.2.0.4, thanks.
